scripts/bumpfee.py

```diff
@@ -0,0 +1,254 @@
+                  'broadcasting the transaction.')
+parser.add_option('-f',
+                  '--txfee',
+                  action='store',
+                  type='int',
+                  dest='txfee',
+                  default=-1,
+                  help='Bitcoin miner tx_fee to use for transaction(s). A number higher '
+                  'than 1000 is used as "satoshi per KB" tx fee. A number lower than that '
+                  'uses the dynamic fee estimation of your blockchain provider as '
+                  'confirmation target. This temporarily overrides the "tx_fees" setting '
+                  'in your joinmarket.cfg. Works the same way as described in it. Check '
+                  'it for examples.')
+parser.add_option('-a',
+                  '--amtmixdepths',
+                  action='store',
+                  type='int',
+                  dest='amtmixdepths',
+                  help='number of mixdepths in wallet, default 5',
+                  default=5)
+parser.add_option('-g',
+                  '--gap-limit',
+                  type="int",
+                  action='store',
+                  dest='gaplimit',
+                  help='gap limit for wallet, default=6',
+                  default=6)
+parser.add_option('--yes',
+                  action='store_true',
+                  dest='answeryes',
+                  default=False,
+                  help='answer yes to everything')
+add_base_options(parser)
+
+def check_valid_candidate(orig_tx, orig_tx_info, wallet):
+    # check that the transaction is still unconfirmed
+    if orig_tx_info['confirmations'] > 0:
+        raise RuntimeWarning('Transaction already confirmed. Nothing to do.')
+
+    # all transaction inputs must belong to the wallet
+    own_inputs_n = len(wallet.inputs_consumed_by_tx(orig_tx))
+    tx_inputs_n = len(orig_tx.vin)
+
+    if own_inputs_n != tx_inputs_n:
+        raise ValueError('Transaction inputs should belong to the wallet.')
+
+    # at least one input should signal opt-in rbf
+    if not any([vin.nSequence <= 0xffffffff - 2 for vin in orig_tx.vin]):
+        raise ValueError('Transaction not replaceable.')
+
+    # wallet should own at least one output that we can change values for
+    if not any(
+        [wallet.is_known_script(vout.scriptPubKey) for vout in orig_tx.vout]
+    ):
+        raise ValueError('Transaction has no output we can deduct fees from.')
+
+    # wallet inputs must be in the same mixdepth - sanity check
+    _mixdepths = set()
+    for ix in range(len(orig_tx.vin)):
+        _mixdepths.add(
+            wallet.get_details(
+                wallet.script_to_path(
+                    wallet.pubkey_to_script(
+                        btc.extract_pubkey_from_witness(orig_tx, ix)[0]
+                    )
+                )
+            )[0]
+        )
+    if len(_mixdepths) != 1:
+        raise ValueError('Transaction inputs must be from the same mixdepth.')
+
+def tx_vsize(tx):
```

Comment:
Yep, this is good, but I think we may as well add it next to `estimate_tx_size` as a utility function in `jmbitcoin.secp256k1_transaction.py` perhaps? Would not be strange to see it useful in another place, too.

Reply:
I'll move it there and call it `estimate_tx_vsize`

Reply:
It's still called `tx_vsize` but that's fine, it's not an estimate.